swap_meet/vendor.py

```diff
@@ -1,2 +1,79 @@
+from .item import Item
+
+# wave 1
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+    
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+#wave 2
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        else:
+            return None
+
+
+#wave 3
+    def swap_items(self,other_vendor, my_item, their_item):
+        if ((my_item not in self.inventory) or
+            (their_item not in other_vendor.inventory)):
```

Comment:
You can safely write this without the parentheses and there's no need to break this line up across two lines because you don't exceed the max line length.
```suggestion
        if my_item not in self.inventory or their_item not in other_vendor.inventory:
```